msg/Result.msg

```diff
@@ -0,0 +1,18 @@
+# Result code and message for service calls.
+# Note that additional results for specific services can defined within them using values above 100.
+
+uint8 FEATURE_UNSUPPORTED   = 0       # Feature is not supported by the simulator, check GetSimulatorFeatures.
+                                      # While feature support can sometimes be deduced from presence of corresponding
+                                      # service / action interface, in other cases it is about supporting certain call
+                                      # parameters, formats and options within interface call.
+uint8 OK                    = 1
+uint8 NOT_FOUND             = 2       # No match for input (such as when entity name does not exist).
+uint8 INCORRECT_STATE       = 3       # Simulator is in an incorrect state for this interface call, e.g. a service
+                                      # requires paused state but the simulator is not paused.
+uint8 OPERATION_FAILED      = 4       # Request could not be completed successfully even though feature is supported and
+                                      # the input is correct; check error_message for details.
+                                      # Implementation rule: check extended codes for called service (e.g. SpawnEntity)
+                                      # to provide a return code which is more specific.
+
+uint8 result                          # Result to be checked on return from service interface call
+string error_message                  # Additional error description when useful.
```

Comment:
The existence of this message seems like a fundamental failing of ROS 2.

Reply:
Could you elaborate? I think I understand what you mean but don't want to jump to conclusions.

Reply:
We discussed this in the PMC meeting, I do think that this `status` message should be a first-class citizen in the ecosystem.  It's very similar to [`absl::Status`](https://abseil.io/docs/cpp/guides/status).  

You will see that many services do this pattern, which indicates to me that it should _at least_ be a common message, but it seems very likely that this could be built into the underlying service mechanism.

Reply:
@mjcarroll That seems very reasonable to me. How do I proceed with the standard? Since the message is clearly not there yet, I can't make a functional message package without including the Result message. Should I keep it in and then before the standard is implemented we would seek to reuse the common message?

Reply:
For now, I included a readme section.